clingwrapper/src/clingwrapper.cxx

```diff
@@ -59,6 +65,7 @@ using namespace CppyyLegacy;
 typedef CPyCppyy::Parameter Parameter;
 // --temp
 
+namespace cling { namespace cppyy { auto *gCling = ((TCling *)gInterpreter)->GetInterpreterImpl(); } }
```

Comment:
You can make `GetInterpreterImpl` protected (which will make it an acceptable patch in ROOT) and here do something like:

```cpp
struct TClingAccessor : public TCling {
  static cling::Interpreter* GetInterp(TInterpreter* This) { return static_cast<TClingAccessor*>(This)->GetInterpreterImpl();}
};
... gCling = TClingAccessor::GetInterp(TCling*(gInterpreter));